### DISCUSSION TITLE
Prevent force-pushing when working with Git remotes (for newbies...)?

### DISCUSSION THREAD

**Discussion by kastl-ars** [jj-vcs/jj#5159] [Q&A]
Hi all,

as I learnt in the documentation, by default `jj` behaves like `git push --force-with-lease` when running `jj git push`.
https://jj-vcs.github.io/jj/latest/bookmarks/#pushing-bookmarks-safety-checks

As I am just starting out using `jj`, I would like to disable this behavior to avoid accidents while getting used to `jj`. Is there a way to do that?

In my understanding this goes against the idea of `jj` doing automatic rebases of child changes etc., which I find very useful. But it will take some time to get my head around that...

Kind Regards,
Johannes

**Comment by gtsiam** [comment]
`jj git push` is similar, but not identical to `git push --force-with-lease`. Mainly:

> The local bookmark must not be [conflicted](https://jj-vcs.github.io/jj/latest/bookmarks/#conflicts). If it is, you would need to use jj bookmark set, for example, to resolve the conflict.

So, as long as you do not explicitly change the relevant bookmark, you are good.

If you want stronger guarantees for "upstream" / "trunk" bookmarks, consider using [immutable commits](https://jj-vcs.github.io/jj/latest/config/#set-of-immutable-commits).

**Reply by kastl-ars** [reply]
Thanks I will have a look!

**Reply by jennings** [reply]
To get you started, if you would like the rule "anything pushed to a remote is immutable", you would use this configuration:

```toml
[revset-aliases]
'immutable_heads()' = 'builtin_immutable_heads() | remote_branches()`
```

The built-in value is `trunk() | tags() | untracked_remote_branches()`, so this simply adds tracked remote branches to the list.

Immutability can be overridden for an individual command by adding `--ignore-immutable`, for example: `jj rebase -d main -b my-branch --ignore-immutable`

**Reply by kastl-ars** [reply]
Thanks a lot!

**Reply by kastl-ars** [reply]
According to the output with the latest jj version, it should be `remote_bookmarks()` instead of `remote_branches()`
